service/matching/tasklist/matcher.go

```diff
@@ -213,15 +213,16 @@ func (tm *TaskMatcher) Offer(ctx context.Context, task *InternalTask) (bool, err
 				task.IsForwarded() { // task came from a child partition
 				// a forwarded backlog task from a child partition, block trying
 				// to match with a poller until ctx timeout
-				return tm.offerOrTimeout(ctx, startT, task)
+				return tm.OfferOrTimeout(ctx, startT, task)
 			}
 		}
 
 		return false, nil
 	}
 }
 
-func (tm *TaskMatcher) offerOrTimeout(ctx context.Context, startT time.Time, task *InternalTask) (bool, error) {
+// OfferOrTimeout offers a task to a poller and blocks until a poller picks up the task or context timeouts
+func (tm *taskMatcherImpl) OfferOrTimeout(ctx context.Context, startT time.Time, task *InternalTask) (bool, error) {
```

Comment:
What was the reason for making this public? I am not sure you need to if you're just doing test coverage?

Reply:
Because I'm setting the matcher as the interface `TaskMatcher` in the `task_list_manager` in order to be able to mock it. But `offerOrTimeout` is called in the `tast_list_manager` from the matcher, and because it implements the interface, it doesn't expose the private methods from the implementation. I can probably define the private method in the interface, but from what I look it's not idiomatic